src/ImageSharp/Formats/Jpeg/JpegEncoderCore.cs

```diff
@@ -466,6 +467,54 @@ private void WriteIptcProfile(IptcProfile iptcProfile)
             this.outputStream.Write(data, 0, data.Length);
         }
 
+        /// <summary>
+        /// Writes the XMP metadata.
+        /// </summary>
+        /// <param name="xmpProfile">The XMP metadata to write.</param>
+        /// <exception cref="ImageFormatException">
+        /// Thrown if the XMP profile size exceeds the limit of 65533 bytes.
+        /// </exception>
+        private void WriteXmpProfile(XmpProfile xmpProfile)
+        {
+            if (xmpProfile is null)
+            {
+                return;
+            }
+
+            const int XmpOverheadLength = 29;
+            const int Max = 65533;
+            const int MaxData = Max - XmpOverheadLength;
+
+            byte[] data = xmpProfile.ToByteArray();
```

Comment:
We can expose an *internal* `byte[] XmpProfile.Data` property to avoid an allocation and copying of the byte array.

Reply:
Makes sense. `XmpProfile` is now effectively read-only from outside, which means we can assume the `data` field can be written directly in all the Encoders.